lib/client/api.go

```diff
@@ -4611,6 +4617,17 @@ func (tc *TeleportClient) NewKubernetesServiceClient(ctx context.Context, cluste
 	return kubeproto.NewKubeServiceClient(clt.GetConnection()), nil
 }
 
+// IsALPNConnUpgradeRequired returns true if connection upgrade is required for
+// provided addr.
+func (tc *TeleportClient) IsALPNConnUpgradeRequired(addr string, insecure bool) bool {
+	// Use cached value.
+	if addr == tc.WebProxyAddr {
+		return tc.TLSRoutingConnUpgradeRequired
+	}
+	// Do a test for other addresses.
+	return client.IsALPNConnUpgradeRequired(addr, insecure)
+}
```

Comment:
Do we need additional flag if the `InsecureSkipVerify` field is already present in the `TeleportClient`  object ? 
```suggestion
func (tc *TeleportClient) IsALPNConnUpgradeRequired(addr string) bool {
	// Use cached value.
	if addr == tc.WebProxyAddr {
		return tc.TLSRoutingConnUpgradeRequired
	}
	// Do a test for other addresses.
	return client.IsALPNConnUpgradeRequired(addr, tc.InsecureSkipVerify)
}
```